_posts/2021/2021-06-17-apis-for-newbies.md

```diff
@@ -0,0 +1,83 @@
+
+After facing this huge lack of knowledge I planned to practice with already given data. At ePages we are following the API first approach, therefore I searched for simple and free useable APIs to get into this topic.
+
+You may have noticed that there is a large pool of APIs out there, but not every API is perfect to start with if you are new to the field. So let me introduce you to my 3 favorite APIs, short and simple enough for beginners.
+
+## The Official Joke API
+
+I would call the [Official Joke API](https://official-joke-api.appspot.com/){:target="_blank"} one of the easiest APIs ever because you don’t need an access token to get the data. This API provides you four different endpoints to get a JSON object of [1 random joke](https://official-joke-api.appspot.com/jokes/random){:target="_blank"} or an array of objects of [10 random jokes](https://official-joke-api.appspot.com/jokes/ten){:target="_blank"}. The only thing left to do is to fetch the data and return the result.
+
+**This is one way how you can do that:**
+
+<iframe src="https://codesandbox.io/embed/fetch-data-from-api-b6vr3?fontsize=14&module=%2Fsrc%2FJokeList.tsx&theme=dark"
+    style="width:100%; height:500px; border:0; border-radius: 4px; overflow:hidden;"
+    title="Fetch data from API"
+    allow="accelerometer; ambient-light-sensor; camera; encrypted-media; geolocation; gyroscope; hid; microphone; midi; payment; usb; vr; xr-spatial-tracking"
+    sandbox="allow-forms allow-modals allow-popups allow-presentation allow-same-origin allow-scripts"
+></iframe>
+
+## The One API (to rule them all)
+
+[The One API](https://the-one-api.dev/){:target="_blank"} is a massive database around the world of Lord of the Rings - I mean, how cool is that? 😱 You get data about the books by J. R. R. Tolkien and about the official movies by Peter Jackson as well as about many characters and quotes. The One API is very well explained and [documented](https://the-one-api.dev/documentation){:target="_blank"} which makes it easy as a newbie to kick-start. You also have the possibility to add pagination, sorting, and filtering options to your API requests. To get access to that awesome database, you need to register and then you get a bearer token. Only the `/book` endpoint is available without authentication. Furthermore, you have a limit to 100 requests every 10 minutes.
+
+**This is how you could fetch the data with your bearer token:**
+
+```js
+const TOKEN = "thisIsYourSecretToken"
+
+const fetchData = async () => {
+  const response = await fetch(
+    `https://the-one-api.dev/v2/character`,
+    {
+      headers: {
+        Authorization: `Bearer ${TOKEN}`
+      }
+    }
+  )
+  const result = await response.json();
+  return result;
+}
```

Comment:
Could you double check if this code works for you with your token?
```suggestion
const fetchData = async () => (
  await fetch(
    `https://the-one-api.dev/v2/character`,
    {
      headers: {
        Authorization: `Bearer ${TOKEN}`
      }
    }
  ).then(response => response.json())
)
```

Reply:
(Unfortunately, I can only test it at a later date.)